languageservice/src/inlay-hints.ts

```diff
@@ -0,0 +1,56 @@
+import {isString} from "@actions/workflow-parser";
+import {getCronDescription} from "@actions/workflow-parser/model/converter/cron";
+import {TemplateToken} from "@actions/workflow-parser/templates/tokens/template-token";
+import {File} from "@actions/workflow-parser/workflows/file";
+import {TextDocument} from "vscode-languageserver-textdocument";
+import {InlayHint, InlayHintKind} from "vscode-languageserver-types";
+import {fetchOrParseWorkflow} from "./utils/workflow-cache.js";
+
+/**
+ * Returns inlay hints for a workflow document.
+ * Currently supports cron expressions, showing a human-readable description
+ * of the schedule inline after the cron value.
+ *
+ * @param document Text document to get inlay hints for
+ * @returns Array of inlay hints
+ */
+export function getInlayHints(document: TextDocument): InlayHint[] {
+  const file: File = {
+    name: document.uri,
+    content: document.getText()
+  };
+
+  const result = fetchOrParseWorkflow(file, document.uri);
```

Comment:
Note, `fetchOrParseWorkflow` uses a shared cache. If validation or hover already parsed the workflow, the cached value is returned.